code/modules/heavy_vehicle/mech_damage.dm

```diff
@@ -37,17 +37,31 @@
 	maxHealth = body.mech_health
 	health = maxHealth-(getFireLoss()+getBruteLoss())
 
-/mob/living/heavy_vehicle/adjustFireLoss(var/amount)
-	var/obj/item/mech_component/MC = pick(list(arms, legs, body, head))
-	if(MC)
-		MC.take_burn_damage(amount)
-		MC.update_health()
-
-/mob/living/heavy_vehicle/adjustBruteLoss(var/amount)
-	var/obj/item/mech_component/MC = pick(list(arms, legs, body, head))
-	if(MC)
-		MC.take_brute_damage(amount)
-		MC.update_health()
+/mob/living/heavy_vehicle/adjustFireLoss(var/amount, var/obj/item/mech_component/C)
+	if(C)
+		C.take_brute_damage(amount)
+		C.update_health()
+	else
+		var/list/components = list(body, arms, legs, head)
+		components = shuffle(components)
+		for(var/obj/item/mech_component/MC in components)
```

Comment:
This will always pick first available item in list, prioritizing body. Shouldn't the order be (legs, body, arms, head)?

Reply:
the shuffle(components) randomizes the list (or should, according to its comments and prior usage) so nah this is fine